components/script/dom/webglprogram.rs

```diff
@@ -31,6 +31,39 @@ pub struct WebGLProgram {
     renderer: WebGLMsgSender,
 }
 
+/// ANGLE adds a `_u` prefix to variable names:
+///
+/// https://chromium.googlesource.com/angle/angle/+/855d964bd0d05f6b2cb303f625506cf53d37e94f
+const ANGLE_NAME_PREFIX: &'static str = "_u";
+
+fn to_name_in_compiled_shader(s: &str) -> String {
+    map_dot_separated(s, |s, mapped| {
+        mapped.push_str(ANGLE_NAME_PREFIX);
```

Comment:
So there's a gotcha here, which is that a name which is close to the name length limit such as the prefix will make it overflow the name length shouldn't be prefixed. From the quote in the mentioned commit:

> An exception needs to be made for identifiers close to the length
limit, since adding any prefix would take them over the limit. But
they can be just written out as such, since we don't have any builtins
or ANGLE internal variables that have as long names and could create a
conflict.

Feel free to not fix it if you don't think it's important enough (I don't think it's worth blocking on that), but please do mention it here.